server/extension/extension.go

```diff
@@ -580,17 +638,16 @@ func (m *Manager) CallExtension(extName string, registry ProxyRegistry) func(htt
 			return
 		}
 
-		sanitizeRequest(r, extName)
+		prepareRequest(r, extName, app)
 		m.log.Debugf("proxing request for extension %q", extName)
 		proxy.ServeHTTP(w, r)
 	}
 }
 
-// sanitizeRequest is reponsible for preparing and cleaning the given
+// prepareRequest is reponsible for preparing and cleaning the given
 // request, removing sensitive information before forwarding it to the
 // proxy extension.
-func sanitizeRequest(r *http.Request, extName string) {
+func prepareRequest(r *http.Request, extName string, app *v1alpha1.Application) {
 	r.URL.Path = strings.TrimPrefix(r.URL.Path, fmt.Sprintf("%s/%s", URLPrefix, extName))
-	r.Header.Del("Cookie")
-	r.Header.Del("Authorization")
+	r.Header.Set(HeaderArgoCDTargetCluster, app.Spec.Destination.Server)
```

Comment:
Apps aren't guaranteed to have the `app.Spec.Destination.Server` field populated. It could be using `app.Spec.Destination.Name` instead. I think we should either send both values on 2 different headers, or get the corresponding URL for the given server name (if set) and use that.

Reply:
fixed